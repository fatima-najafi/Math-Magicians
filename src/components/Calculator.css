* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.calculator {
  display: flex;
  flex-direction: column;
  margin-left: 50%;
  width: 400px;
  margin-bottom: 50px;
}

.text-p {
  font-size: 22px;
  margin-left: 200px;
  margin-top: 60px;
}

.display {
  width: 100%;
  height: 60px;
  display: flex;
  flex-direction: row;
  justify-content: right;
  align-items: center;
  background-color: #e3dfdf;
  font-size: 30px;
  padding: 17px;
  text-align: right;
  border: none;
  outline: none;
  border-radius: 10px;
  box-shadow: 5px 5px 15px rgb(0, 0, 0.2);
}

.parent {
  display: grid;
  grid-template-columns: auto auto auto auto;
  background-color: #e3dfdf;
  border-radius: 5px;
  box-shadow: 5px 5px 15px rgb(0, 0, 0.2);
}

.button {
  background-color: rgba(199, 199, 199, 0.995);
  padding: 20px;
  font-size: 22px;
  text-align: center;
  margin: 1px;
  width: 98px;
  height: 80px;
  border: 0;
}

.button-zero {
  grid-column: 1/3;
  background-color: rgba(199, 199, 199, 0.995);
  border: 0;
  font-size: 22px;
}

.button-color {
  background-color: orange;
  padding: 20px;
  font-size: 15px;
  text-align: center;
  margin: 1px;
  width: 98px;
  height: 80px;
  border: 0;
}
