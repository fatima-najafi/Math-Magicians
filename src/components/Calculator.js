import React from 'react';
import './Calculator.css';
import ButtonDiv from './ButtonDiv';

function Calculator() {
  return (
    <div className="calculator">
      <div>
        <input type="text" className="input" id="name" placeholder="0" />
        <ButtonDiv />
      </div>
    </div>
  );
}
export default Calculator;
