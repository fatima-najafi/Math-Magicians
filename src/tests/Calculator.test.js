import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('render the calculator', () => {
  test('displays the calculator text', () => {
    const { container } = render(<Calculator />);
    const textCalculator = screen.getByText(/Lets do some math!/i);
    expect(textCalculator).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
