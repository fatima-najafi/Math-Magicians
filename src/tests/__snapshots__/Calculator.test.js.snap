// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the calculator displays the calculator text 1`] = `
<div>
  <div
    class="con-container"
  >
    <p
      class="text-p"
    >
      Lets do some math!
    </p>
    <div
      class="calculator"
    >
      <div
        class="display"
      />
      <div
        class="parent"
      >
        <button
          class="button"
          type="button"
        >
          AC
        </button>
        <button
          class="button"
          type="button"
        >
          +/-
        </button>
        <button
          class="button"
          type="button"
        >
          %
        </button>
        <button
          class="button-color"
          type="button"
        >
          /
        </button>
        <button
          class="button"
          type="button"
        >
          7
        </button>
        <button
          class="button"
          type="button"
        >
          8
        </button>
        <button
          class="button"
          type="button"
        >
          9
        </button>
        <button
          class="button-color"
          type="button"
        >
          *
        </button>
        <button
          class="button"
          type="button"
        >
          4
        </button>
        <button
          class="button"
          type="button"
        >
          5
        </button>
        <button
          class="button"
          type="button"
        >
          6
        </button>
        <button
          class="button-color"
          type="button"
        >
          -
        </button>
        <button
          class="button"
          type="button"
        >
          1
        </button>
        <button
          class="button"
          type="button"
        >
          2
        </button>
        <button
          class="button"
          type="button"
        >
          3
        </button>
        <button
          class="button-color"
          type="button"
        >
          +
        </button>
        <button
          class="button-zero"
          type="button"
        >
          0
        </button>
        <button
          class="button"
          type="button"
        >
          .
        </button>
        <button
          class="button-color"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
